### PR TITLE
feat: add support for more diagrams

### PR DESCRIPTION
**Pull request by oscaramos**
Related to #3 

## Description
This pull request enhances the functionality of the project by enabling the use of all languages supported by Kroki. While I observed that certain languages outperform others, I have instructed the model to prioritize the use of the mermaid language.

I have refactored the codebase to facilitate the addition of language-related functions for the newly supported languages in Kroki. Functions like diagram formatting, obtaining the editor link, and validating syntax.

In my opinion, most of the times, the diagrams look great even without prompt engineering them for each language.

### Show me how a food critic can interact with a restaurant using plantuml
image

### Illustrate a sequence diagram of an HTTP request and response involving a user, browser, and a server. Use plantuml.
image

### Draw me a family tree of 3 generations, one child at the bottom and many grandparents at the top. Use the most popular names in the world, use graphviz.
image

### Make a linked list of prime numbers from 5 to 29 with blockdiag and make each node a different color like a rainbow
image

 
### Draw ascii art of three persons with names above their heads using svgbob
image

### Create a database diagram of social media like twitter with 4 tables using dbml
image

### CODE REVIEW

File: src/routes/Diagram.ts
Comment (bra1nDump) [inline src/routes/Diagram.ts:222]: We can kill this code, probably will not be used ever

File: src/routes/Diagram.ts
Comment (bra1nDump) [inline src/routes/Diagram.ts:139]:
```suggestion
            errorMessage: "GPT created an invalid diagram, you can try again or edit it online",
```

File: src/routes/Diagram.ts
Comment (bra1nDump) [inline src/routes/Diagram.ts:119]:
```suggestion
```

File: src/routes/Diagram.ts
Comment (bra1nDump) [reply src/routes/Diagram.ts:222]: We can merge without the cleanup part, I can clean it up after we merge this if you want

File: src/routes/diagrams/index.ts
Comment (bra1nDump) [inline src/routes/diagrams/index.ts:10]: Since we are already fetching SVG to check if its valid can we also add svg here?
Then we can do the #1 in a straight forward way without using the API 2ce.

**Review by bra1nDump — CHANGES_REQUESTED** [review]
Love the changes!
Great new abstraction with diagram details.

Let me know if you have time to make the changes.

Also regarding the new release to production: 
Last time I tried updating the plugin I have taken it down from the plugin store so I want to make sure we have guidance from OpenAI on whats the best way to do updates so we don't take the old version down.

Also to avoid breaking clients who were using the plugin while we are rolling the update (even the official OpenAI way whatever that would be) we should have backwards compatible API. Basically keeping the old parameter in place for the time being: diagrams and converting that case to the new format.

Grateful for your help!

File: src/routes/Diagram.ts
Comment (oscaramos) [reply src/routes/Diagram.ts]: Done!

File: src/routes/diagrams/index.ts
Comment (oscaramos) [reply src/routes/diagrams/index.ts]: I have resolved issue #1 by replacing the imageURL with diagramSVG. I see the increased speed.

**Comment by bra1nDump** [comment]
🙌🙌

File: src/routes/diagrams/utils.ts
Comment (oscaramos) [reply src/routes/diagrams/utils.ts:76]: I checked the languages used in the Examples section and I see that the checking function covers all the cases.

### The "Syntax error in graph" when there is a syntax error in mermaid:

image

### On others it returns HTTP failing codes
image

I did not have the time to test all languages but I think that the code will handle at least the majority

**Pull request merged**
